code/game/mecha/equipment/tools/other_tools.dm

```diff
@@ -434,6 +434,41 @@
 	if(..())
 		radiation_pulse(get_turf(src), rad_per_cycle)
 
+/obj/item/mecha_parts/mecha_equipment/thrusters
+	name = "exosuit ion thrusters"
+	desc = "Ion thrusters to be attached to an exosuit. Drains power even while not in flight."
+	icon_state = "tesla"
+	origin_tech = "powerstorage=4;engineering=4"
+	range = 0
+	energy_drain = 20
+	selectable = FALSE
+
+/obj/item/mecha_parts/mecha_equipment/thrusters/attach(obj/mecha/M)
+	. = ..()
+	START_PROCESSING(SSobj, src)
+	M.add_thrusters()
+	M.thruster_count++
+
+/obj/item/mecha_parts/mecha_equipment/thrusters/detach(atom/moveto)
+	chassis.remove_thrusters()
```

Comment:
this should probably only be called if thruster count is going to drop to 0

Reply:
applied